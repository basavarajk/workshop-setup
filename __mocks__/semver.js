module.exports = {
  satisfies: jest.fn(),
}
