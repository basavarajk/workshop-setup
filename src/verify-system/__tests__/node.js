import mockProcess from 'process'
import {satisfies as satisfiesMock} from 'semver'
import validateNode from '../node'

jest.mock('process', () => ({
  versions: {
    node: '6.9.5',
  },
}))

test('returns null if conditions are satisfied', () => {
  setup({version: '6.9.5'})
  const desired = '6.9.5'
  const result = validateNode(desired)()
  expect(result).toBe(null)
})

test('returns error if conditions are not satisfied', () => {
  const mockActual = '4.7.0'
  setup({version: mockActual, satisfies: false})
  const desired = '6.9.5'
  const result = validateNode(desired)()
  expect(result).toContain(desired)
  expect(result).toContain(mockActual)
  expect(result).toMatchSnapshot()
})

function setup({version = '6.9.5', satisfies = true} = {}) {
  mockProcess.versions.node = version
  satisfiesMock.mockClear()
  satisfiesMock.mockImplementation(() => satisfies)
}
