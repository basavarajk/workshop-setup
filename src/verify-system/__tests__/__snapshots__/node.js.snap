// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns error if conditions are not satisfied 1`] = `"The installed version of node (4.7.0) does not satisfy the desired range of 6.9.5. Please install a version within the range. You can use http://git.io/nvm or https://github.com/coreybutler/nvm-windows to make changing The installed version of node easier."`;
